server/src/main/java/com/networknt/server/Server.java

```diff
@@ -269,7 +273,7 @@ static private boolean bind(HttpHandler handler, int port) {
             server.start();
             System.out.println("HOST IP " + System.getenv(STATUS_HOST_IP));
         } catch (Exception e) {
-            if (!serverConfig.dynamicPort || bindCounter > (serverConfig.maxPort - serverConfig.minPort)) {
+            if (!serverConfig.dynamicPort || usedPorts.size() >= (serverConfig.maxPort - serverConfig.minPort)) {
```

Comment:
usedPorts.size() >= (serverConfig.maxPort - serverConfig.minPort) + 1 would be more accurate？Since the final size of usedPort should equal to the capacity.

Reply:
@jiachen1120 I tried your code with +1 and the server won't quit when running out of port. I am using min 2000 and max 2003 for testing on my local with the service discovery consul folder. Let me debug into it. Thanks.

Reply:
When usedPorts is 3, you have tried 4 ports already, the last try is not reflected in the hashset yet.